command/operator_diagnose_test.go

```diff
@@ -21,7 +21,6 @@ func testOperatorDiagnoseCommand(tb testing.TB) (*cli.MockUi, *OperatorDiagnoseC
 }
 
 func TestOperatorDiagnoseCommand_Run(t *testing.T) {
-	t.Parallel()
```

Comment:
These parallels cause listeners to collide with one another if they share ports (eg 2 tests using port 1024 will cause a race failure in which either test will randomly fail with a "port already in use" error). So I removed them, allowing listeners to be cleaned up between runs via the defer listener.close block.

Reply:
Could you assign them each a different incrementing port #?

Reply:
Oh, possibly not since we don't control the port #s

Reply:
We do control the port numbers, and I could assign different port numbers (initially I did that to bypass this problem), but I changed it to use the same port and remove the parallel. I don't like the idea that these tests are going to open a bunch of listeners at the same time on different ports, and removing parallel fixes that issue. 

Also it seems like using Parallel() in our tests is an anti-pattern that we want to avoid.

Reply:
Parallel is actually good so long as it doesn't change the semantics of the tests, helps CI run more quickly.

Reply:
I spoke too soon! Parallel and different port numbers may be the way to go here after all!